src/test/java/com/synopsys/integration/stepworkflow/StepWorkflowTest.java

```diff
@@ -6,26 +6,65 @@
 import static org.junit.jupiter.api.Assertions.assertNull;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
 
 import com.synopsys.integration.exception.IntegrationException;
 
 public class StepWorkflowTest {
-    public static final Integer DATA = 1;
+    public static final Integer DATA_ONE = 1;
+    public static final Integer DATA_TWO = 2;
+    public static final Integer DATA_THREE = 3;
+    public static final Integer DATA_FOUR = 4;
+
+    private static final IntegrationException integrationException = new IntegrationException("Exception #1");
+
+    private static final SubStep<Object, Integer> subStepSupplierDataOne = SubStep.ofSupplier(() -> DATA_ONE);
+    private static final SubStep<Object, Integer> subStepSupplierDataTwo = SubStep.ofSupplier(() -> DATA_TWO);
+    private static final SubStep<Object, Integer> subStepSupplierDataThree = SubStep.ofSupplier(() -> DATA_THREE);
+    private static final SubStep<Integer, Object> subStepFunctionDataFour = SubStep.ofFunction((Integer integer) -> DATA_FOUR);
```

Comment:
Is there any reason for this to be a function instead of a supplier? It looks like you're just using this as a step like the others above it.

Reply:
Updated.